core/utils.c

```diff
@@ -417,7 +417,8 @@ locks_not_closed()
                     cur_lock->rank ==
                         LOCK_RANK(options_lock)
                         /* This lock can be used parallel to detach cleanup. */
-                        IF_UNIX(|| cur_lock->rank == LOCK_RANK(detached_sigact_lock)))) {
+                        IF_UNIX(|| cur_lock->rank == LOCK_RANK(detached_sigact_lock))
+                            IF_LINUX(|| cur_lock->rank == LOCK_RANK(dump_core_lock)))) {
```

Comment:
This new lock should be cleaned up: please add a destroy instead of an exception.

Reply:
The lock is deleted in os_fast_exit in win32/os.c. is os_slow_exit in unix/os.c the right please to delete it?

Reply:
win32/os.c is windows-only code never executed for non-windows.

Reply:
Oh I see, this lock already existed but only on Windows.  You added an identical decl in unix/coredump.c.  Hmm.  Seems cleaner to either move the decl and cleanup to a shared file, or have two separate locks with two separate names?

Reply:
I've updated the code to use dr_suspend_all_other_threads_ex(). I've reverted changes related to dump_core_lock.
PTAL.